src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -334,16 +334,53 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
     * @param requestContext RequestContext
     * @param userInfo       UserInfo
     * @param spreadsheetId  Spreadsheet ID
-    * @param content        ValueRange
+    * @param newContent     ValueRange
     * @return               JsObject representing the Google Update response
     */
-  def updateSpreadsheet(requestContext: RequestContext, userInfo: UserInfo, spreadsheetId: String, content: ValueRange): JsObject = {
+  def updateSpreadsheet(requestContext: RequestContext, userInfo: UserInfo, spreadsheetId: String, newContent: ValueRange): JsObject = {
     val credential = new GoogleCredential().setAccessToken(userInfo.accessToken.token)
     val service = new Sheets.Builder(httpTransport, jsonFactory, credential).setApplicationName("FireCloud").build()
-    val response = service.spreadsheets().values().update(spreadsheetId, content.getRange, content).setValueInputOption("RAW").execute()
+
+    // Retrieve existing records
+    val getExisting: Sheets#Spreadsheets#Values#Get = service.spreadsheets().values.get(spreadsheetId, "Sheet1")
+    val existingContent: ValueRange = getExisting.execute()
+
+    // Smart merge existing with new
+    val rows = updatePreservingOrder(newContent, existingContent)
+
+    // Send update
+    val response = service.spreadsheets().values().update(spreadsheetId, newContent.getRange, newContent.setValues(rows)).setValueInputOption("RAW").execute()
     response.toString.parseJson.asJsObject
   }
 
+  private def updatePreservingOrder(newContent: ValueRange, existingContent: ValueRange): List[java.util.List[AnyRef]] = {
+    val header: java.util.List[AnyRef] = existingContent.getValues.get(0)
+    val existingRecords: List[java.util.List[AnyRef]] = existingContent.getValues.drop(1).toList
+
+    val newRecords: List[java.util.List[AnyRef]] = newContent.getValues.drop(1).toList
+
+    // Go through existing records and update them in place
+    val existingRecordsUpdated = existingRecords.map { existingRecord =>
+      val matchingNewRecord = newRecords.find { newRecordCandidate =>
+        newRecordCandidate.get(0) == existingRecord.get(0)
+      }
+
+      matchingNewRecord match {
+        case Some(newRecord) => newRecord // Overwrite the entire row in place, adequate unless we want to preserve user-added columns
+        case _ => existingRecord // Don't delete existing records that aren't in the new export
+      }
+    }
+
+    // Create new records for newly-added projects
+    val recordsToAppend = newRecords.filter { newRecordCandidate =>
+      !existingRecords.exists { existingRecordCandidate =>
+        existingRecordCandidate.get(0) == newRecordCandidate.get(0)
+      }
+    }
+
+    List(header) ++ existingRecordsUpdated ++ recordsToAppend
+  }
+
```

Comment:
I would have transformed both `newRecords` and `existingRecords` into maps keyed by the project id: `Map[String, List[AnyRef]]`. Then, inside your loops, instead of doing a `find` or `exists`, you can do a more-efficient map lookup.

At the scale we're talking about - getting up to a thousand-ish rows - this may be premature optimization. So don't consider it a requirement for this PR; I'm mentioning it for the sake of talking about code :)

Reply:
+1 to this idea.

Reply:
This is true, I was lazy and optimized for simplicity on the first pass.

The above would need things like `record.head.asInstanceOf[String]` to wrangle row data from `java.util.List[java.util.List[Object]]` into the right types, and I kinda felt like leaving them alone